### DISCUSSION TITLE
Vue: custom formatting of display columns

### DISCUSSION THREAD

**Discussion by JuliusSettili** [TanStack/table#4640] [Q&A] [answered]
Hello everyone,
I just started working with `@tanstack/vue-table` and tried to implement a simple display column like so:
```typescript
columnHelper.display({
    id: 'actions',
    cell: (cellProps) => `<span>${cellProps.row.getValue('id')}</span>`,
  })
```

But unfortunately the html is not rendered, but is displayed as a plain string:
```html
<td>
    "<span>2</span>"
</td>
```

How can I get the html to render, without using tsx syntax like in all the examples in the docs?

**Comment by v1adko** [comment]
Same question. How do you create dynamic cells? Documentation and examples for Vue is quite lacking, and it seems like a fundamental feature.

**Comment by JuliusSettili** [comment, marked as answer]
In the meantime I found a solution:
```
cell: (cellProps) => cellProps.getValue().map((value) => h(
      'span',
      value,
    )),
```

**Reply by Kolobamanacas** [reply]
> In the meantime I found a solution:
> 
> ```
> cell: (cellProps) => cellProps.getValue().map((value) => h(
>       'span',
>       value,
>     )),
> ```

Unfortunately, it's far from perfect solution as I guess we all want to have possibility to pass custom Vue components with their styles, classes and props. Did you find a way to implement this by any chance?

**Reply by seanockert** [reply]
You can pass any HTML params to the `h` render function like so: 

```js
{
  header: 'Email',
  accessorKey: 'email',
  cell: (cellProps: any) => h('a', { href: `mailto:${cellProps.getValue()}`, class: 'some-class' }, cellProps.getValue())
}
```

this wraps the email address in each row cell: 
```js
<td>
  <a href="mailto:me@example.com" class="some-class">me@example.com</a>
</td>
```

You can use Vue components in the same way eg. if you want to wrap a status in a component called `Tag`:

```js
import { h } from 'vue'
import Tag from 'Tag.vue'

...

const columns = [
  {
    header: 'Status',
    accessorKey: 'status',
    cell: (cellProps: any) => h(Tag, { id: 1, status: 'success' }, cellProps.getValue())
  },
  ...
]
```

I've been using Tanstack Table on a few Vue3 projects and the above works pretty well. There's no need for the extra `map` function as mentioned in the comment above

**Reply by Kolobamanacas** [reply]
Thanks a lot for the examples! I didn't know about the `h` function before. )

**Comment by bloor** [comment]
Yea but using h renderer is obnoxious ... You mean I have to create a component each time I want a customized cell ? eh??!
Isn't there any way to directly use `JSX` or `Template as string` what am I missing ?
